StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement+Internal.swift

```diff
@@ -331,14 +331,63 @@ extension EmbeddedPaymentElement: EmbeddedFormViewControllerDelegate {
         // If the formViewController was populated with a previous payment option don't reset
         if embeddedFormViewController.previousPaymentOption == nil {
             embeddedPaymentMethodsView.resetSelectionToLastSelection()
+            // Show change button if the newly selected row needs it
+            if let newSelectedType = embeddedPaymentMethodsView.selectedRowButton?.type {
+                let changeButtonState = getChangeButtonState(for: newSelectedType)
+                if changeButtonState.shouldShowChangeButton {
+                    embeddedPaymentMethodsView.selectedRowButton?.addChangeButton(sublabel: changeButtonState.sublabel)
+                }
+            }
         }
         embeddedFormViewController.dismiss(animated: true)
     }
 
     func embeddedFormViewControllerDidContinue(_ embeddedFormViewController: EmbeddedFormViewController) {
+        // Show change button if the selected row needs it
+        if let newSelectedType = embeddedPaymentMethodsView.selectedRowButton?.type {
+            let changeButtonState = getChangeButtonState(for: newSelectedType)
+            if changeButtonState.shouldShowChangeButton {
+                embeddedPaymentMethodsView.selectedRowButton?.addChangeButton(sublabel: changeButtonState.sublabel)
+            }
+        }
         embeddedFormViewController.dismiss(animated: true)
         informDelegateIfPaymentOptionUpdated()
     }
+    
+    func getChangeButtonState(for type: RowButtonType) -> (shouldShowChangeButton: Bool, sublabel: String?) {
+        guard let _paymentOption, let displayData = paymentOption else {
+            return (false, nil)
+        }
+        // Show change button for new PMs that have a valid form
+        let shouldShowChangeButton: Bool = {
+            if case .new = type, selectedFormViewController != nil {
+               return true
+            }
+            return false
+        }()
+        
+        // Add a sublabel to the selected row for cards and us bank account like "Visa 4242"
+        let sublabel: String? = {
+            switch type.paymentMethodType {
+            case .stripe(.card):
+                guard
+                    case .new(confirmParams: let params) = _paymentOption,
+                    let cardNumber = params.paymentMethodParams.card?.number
+                else {
+                    return nil
+                }
+                let brand = STPCardValidator.brand(forNumber: cardNumber)
+                let brandString = STPCardBrandUtilities.stringFrom(brand)
```

Comment:
This doesn't seem to work with co-branded cards, when I enter a co-branded (Visa/CB) card and choose CB, the copy says "Visa ...". You'll need to also check the preferred network (if it exists) on the card params.

`params.paymentMethodParams.card?.networks?.preferred`

There's also a weird edge case where you can fill out a form for a co-branded card and don't select a brand since it's optional, so the card brand gets decided on the backend.

Reply:
Thanks for catching this! Fixed it up. Are there things we can do to make it harder to make this mistake?

Reply:
We did something similar to what you did with `func brand(for card: STPPaymentMethodCardParams?) -> STPCardBrand` on `STPPaymentMethodCard`. But this is the first instance AFAIK where we have needed to read it off the `STPPaymentMethodCardParams`.  Maybe we "internally" deprecate STPCardBrandUtilities.stringFrom(brand) somehow, since the card number is not the single source of truth for the card brand for a number.